### ISSUE TITLE
Upgrade Celery to 5.x

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#17539]
We chose to stay on Celery LTS in https://github.com/mozilla/addons-server/issues/15677 but no new LTS has been announced and the version we're using will be EOL soon, so we need to switch to 5.x.

This means unblocking pinned dependencies, ensuring compatibility of things like https://github.com/mozilla/django-post-request-task/, and of course getting everything to work and tests to pass.

**Comment by diox** [comment]
To prepare for that I've released django-post-request-task 0.5 to PyPI. I've only tested it locally with Celery 4 for the moment but its unit tests pass with Celery 5 now.

**Comment by diox** [comment]
Bumping priority since the EOL of celery 4 is near.

**Comment by diox** [comment]
QA: test that add-on upload and reindexing for search after changes work.

**Comment by ioanarusiczki** [comment]
@diox I tested today the reindex for search after making changes: 
- when I'm authorized and changed: name, summary, ratings, icons, authors, deleted/disabled/invisible - the changes are immediately visible into the search results.  
- unauthorized users might see the changes with a delay (except for icons and authors). For example, if an addon was deleted, it will take a while until it's not displayed on the search results page (I had to make several page refreshes and wait a while). I think it's a result of the cache feature.
